Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -978,18 +975,37 @@ Optional<HitTestResult> PaintableBox::hit_test(CSSPixelPoint position, HitTestTy
 {
     Optional<HitTestResult> result;
     (void)PaintableBox::hit_test(position, type, [&](HitTestResult candidate) {
-        if (candidate.paintable->visible_for_hit_testing()) {
-            if (!result.has_value()
-                || candidate.vertical_distance.value_or(CSSPixels::max_integer_value) < result->vertical_distance.value_or(CSSPixels::max_integer_value)
-                || candidate.horizontal_distance.value_or(CSSPixels::max_integer_value) < result->horizontal_distance.value_or(CSSPixels::max_integer_value)) {
-                result = move(candidate);
-            }
+        if (!result.has_value()
+            || candidate.vertical_distance.value_or(CSSPixels::max_integer_value) < result->vertical_distance.value_or(CSSPixels::max_integer_value)
+            || candidate.horizontal_distance.value_or(CSSPixels::max_integer_value) < result->horizontal_distance.value_or(CSSPixels::max_integer_value)) {
+            result = move(candidate);
         }
 
         if (result.has_value() && (type == HitTestType::Exact || (result->vertical_distance == 0 && result->horizontal_distance == 0)))
             return TraversalDecision::Break;
         return TraversalDecision::Continue;
     });
+
+    // If our hit-testing has resulted in a hit on a paintable, we know that it is the most specific hit. If that
+    // paintable turns out to be invisible for hit-testing, we need to traverse up the paintable tree to find the next
+    // paintable that is visible for hit-testing. This implements the behavior expected for pointer-events.
+    while (result.has_value() && !result->paintable->visible_for_hit_testing()) {
+        result->index_in_node = result->paintable->dom_node() ? result->paintable->dom_node()->index() : 0;
+        result->paintable = result->paintable->parent();
+
+        // If the new parent is part of a continuation, we need to follow the chain to the original inline node.
```

Comment:
could you add a bit more clarification here why we can't stop on a paintable created by continuation and need to follow up until the origin node?

Reply:
> could you add a bit more clarification

Absolutely, expanded the second comment in that piece of code.

> why we can't stop on a paintable created by continuation

We could actually be more specific: if we end up on an anonymous wrapper that is part of a continuation chain, this is always a "middle" part (since the "before" and "after" parts are inline nodes) and we need to follow the chain all the way to the "before" part to find the actual parent node. Otherwise, we would return a hit result on the wrapper.